hydrocron/api/controllers/timeseries.py

```diff
@@ -258,9 +258,17 @@ def lambda_handler(event, context):  # noqa: E501 # pylint: disable=W0613
     """
 
     start = time.time()
-    print(f"Event - {event}")
+    print(f'Event - {event}')
 
     results = {'http_code': '200 OK'}
+
+    try:
+        if event['body'] == {} and 'Elastic-Heartbeat' in event['headers']['User-Agent']:
+            return {}
+    except KeyError as e:
+        print(f'Error encountered with headers: {e}')
+        raise RequestError('400: Issue encountered with request headers') from e
```

Comment:
I think the only way this can happen is if the `User-Agent` header is not provided, correct? We should be more specific in the error shown to the user so they can try to correct it. 

The exception stacktrace should be logged but not displayed to the user. How does RequestError get translated to the client?

Reply:
This would happen if the `User-Agent` and `X-Forwarded-For` headers are not provided as the `X-Forwarded-For` header is used in issue #125.  I don't think this is likely to happen very often but it would be good to know if the request headers passed to the Lambda were invalid.

I updated the code to display the header back to the user in the error response but was unable to test this in SIT as when I removed the `User-Agent` header from the request it was replaced by CloudFront.

The exception stacktrace is logged to CloudWatch and not displayed to the user. The RequestError would get translated to the client but has only been tested locally as headers are replaced by CloudFront in the SIT environment. Error response:

```json
{
  "error" : "400: Issue encountered with request header: 'User-Agent'"
}
```
I think it is pretty unlikely that we would run into a header issue but wanted to leave the check in just in case something went weird between the API gateway and Lambda. I am happy to take it out if it is just adding complexity that we don't need.

Reply:
Nope, I think this is better. My main concern was categorizing this as a 4xx client error but not giving the client enough information to know how to fix it. Adding the specific header name to the error message I think at least gives the client a hint about where to look.